### ISSUE TITLE
Add int16 support to the CMSIS-NN version of softmax

### ISSUE THREAD

**Issue by jenselofsson** [tensorflow/tensorflow#43543]
@tensorflow/micro

The softmax reference kernel have been updated with int16 support, and the CMSIS-NN version should to updated with int16 support as well.

**Comment by jenselofsson** [comment]
Fixed in #43544

**Comment by jenselofsson** [comment]
Sorry for the confusion, 43545 is the correct PR
